avocado/utils/network/interfaces.py

```diff
@@ -314,3 +314,29 @@ def remove_ipaddr(self, ipaddr, netmask):
         except Exception as ex:
             msg = 'Failed to remove ipaddr. {}'.format(ex)
             raise NWException(msg)
+
+    def move_interface_file(self, ignore_missing=True):
+        """
+        Revert interface file from backup
+
+        This method checks if a backup version  is available for given interface
+        then it copies backup file to interface file in /sysfs path
+        """
+
+        current_distro = distro_detect()
+
+        filename = "ifcfg-{}".format(self.name)
+        if current_distro.name in ['rhel', 'fedora']:
+            path = "/etc/sysconfig/network-scripts"
+        elif current_distro.name == 'SuSE':
+            path = "/etc/sysconfig/network"
+        else:
+            msg = 'Distro not supported by API. Could not save ipaddr.'
+            raise NWException(msg)
+
+        backup_file = "{}.backup-{}".format(filename, time.time())
+        if os.path.exists(backup_file):
+            shutil.move(backup_file, filename)
+        else:
+            if not ignore_missing:
+                raise NWException("%s interface not available" % self.name)
```

Comment:
"Backup file not available, could not restore file."

Also, check my previous comment about `ignore_missing`. IMO, we don't need it.

Reply:
ah miss that

Reply:
@PraveenPenguin just to be clear, we don't need the `ignore_missing` flag, but still it is good to have a `NWException` always when the file is not found.